server/src/main/java/org/opensearch/indices/IndicesService.java

```diff
@@ -513,6 +515,10 @@ private void onRefreshIntervalUpdate(TimeValue clusterDefaultRefreshInterval) {
         }
     }
 
+    public Supplier<RepositoriesService> getRepositoriesServiceSupplier() {
+        return repositoriesServiceSupplier;
+    }
```

Comment:
rather than making this public, can we pass the `repositoriesServiceSupplier` in the ctor of MetadataCreateIndexService?